conformance/utils/kubernetes/helpers.go

```diff
@@ -152,3 +155,104 @@ func InferencePoolMustHaveCondition(t *testing.T, c client.Client, poolNN types.
 	}
 	t.Log(logMsg)
 }
+
+// FindCondition finds a condition in a list of conditions.
+func FindCondition(conditions []metav1.Condition, conditionType string) *metav1.Condition {
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			return &conditions[i]
+		}
+	}
+	return nil
+}
+
+// HTTPRouteMustHaveParentStatusConditions waits for an HTTPRoute to have specific conditions
+// on its parent status for a given Gateway, identified by its name, namespace, and controller.
+// It checks for an "Accepted" condition and another "Failure" condition (e.g., "Reconciled" or "ResolvedRefs")
+// with expected status, reason, and message substrings.
+func HTTPRouteMustHaveParentStatusConditions(
+	t *testing.T,
+	c client.Client,
+	routeNN types.NamespacedName,
+	gatewayNN types.NamespacedName,
+	controllerName string,
+	expectedAcceptedCond metav1.Condition, // Should have Type and Status. Reason is optional.
+	expectedFailureCond metav1.Condition, // Should have Type, Status, and Reason.
+	expectedFailureMessageSubstrings []string,
```

Comment:
We're not prescriptive in the API spec about what a message should include, so I'd recommend leaving this out of the test.

Reply:
Acknowledged.